### PR TITLE
Add NUMBER keyword

### PR DESCRIPTION
**Pull request by fivetran-dmitryboykov**
This PR introduces NUMBER keyword as a synonym for NUMERIC and DECIMAL types.

### CODE REVIEW

File: zetasql/analyzer/resolver_test.cc [hunk 1]
```diff
@@ -718,7 +718,7 @@ TEST_F(ResolverTest, TestResolveCastExpression) {
   ResolveFunctionFails("CAST(1 as BIGINT)", "Type not found: BIGINT");
   ResolveFunctionFails("CAST(1 as SMALLINT)", "Type not found: SMALLINT");
   ResolveFunctionFails("CAST(1 as real)", "Type not found: real");
-  ResolveFunctionFails("CAST(1 as NUMBER)", "Type not found: NUMBER");
+//   ResolveFunctionFails("CAST(1 as NUMBER)", "Type not found: NUMBER");
```
Comment (pgrivachev) [inline zetasql/analyzer/resolver_test.cc:721]: Interesting that there was a test for this type.

**Review by pgrivachev — APPROVED** [review]
Looks good. Can we add some test for this keyword?

File: zetasql/analyzer/resolver_test.cc (hunk 1 above)
Comment (fivetran-dmitryboykov) [reply zetasql/analyzer/resolver_test.cc:721]: It seems that these types are not supported in Google Standard SQL.

**Review by pgrivachev — APPROVED** [review]
Great! If it possible, I'd suggest to move snow specific tests to separate files.

**Pull request merged**